test_prefect_ensemble.py test_prefect_no_retries is flaky

The test arbitrarily failed when running komod unit tests for ert. In 2021.06.rc4. When re-running it, it did not fail. But we should investigate this.
```
=================================== FAILURES ===================================
___________________________ test_prefect_no_retries ____________________________

unused_tcp_port = 34978
coefficients = [{'a': 1, 'b': 2, 'c': 3}, {'a': 4, 'b': 2, 'c': 1}]
tmpdir = local('/tmp/pytest-of-f_scout_ci/pytest-726/test_prefect_no_retries0')
function_config = {'dispatch_uri': 'wss://10.217.112.232:34978/dispatch', 'executor': 'local', 'inputs': {0: {'coeffs': <ert3.data._reco...780>}, 1: {'coeffs': <ert3.data._record.SharedDiskRecordTransmitter object at 0x7fa226ebe2b0>}}, 'max_retries': 0, ...}

    @pytest.mark.timeout(60)
    def test_prefect_no_retries(unused_tcp_port, coefficients, tmpdir, function_config):
        def function_that_fails_once(coeffs):
            run_path = Path("ran_once")
            if not run_path.exists():
                run_path.touch()
                raise RuntimeError("This is an expected ERROR")
            run_path.unlink()
            return []
    
        with tmpdir.as_cwd():
            pickle_func = cloudpickle.dumps(function_that_fails_once)
            config = function_config
            coeffs_trans = coefficient_transmitters(
                coefficients, config.get(ids.STORAGE)["storage_path"]
            )
    
            service_config = EvaluatorServerConfig(unused_tcp_port)
            config["realizations"] = len(coefficients)
            config["executor"] = "local"
            config["max_retries"] = 0
            config["retry_delay"] = 1
            config["steps"][0]["jobs"][0]["executable"] = pickle_func
            config["inputs"] = {
                iens: coeffs_trans[iens] for iens in range(len(coefficients))
            }
            config["outputs"] = output_transmitters(config)
            config["dispatch_uri"] = service_config.dispatch_uri
    
            ensemble = PrefectEnsemble(config)
            evaluator = EnsembleEvaluator(ensemble, service_config, 0, ee_id="1")
    
            step_failed = False
            job_failed = False
            with evaluator.run() as mon:
                for event in mon.track():
                    # Capture the job error messages
                    if event.data is not None and "This is an expected ERROR" in str(
                        event.data
                    ):
                        for real in event.data["reals"].values():
                            for step in real["steps"].values():
                                for job in step["jobs"].values():
                                    if job["status"] == "Failed":
                                        job_failed = True
                                        if step["status"] == "Failed":
                                            step_failed = True
    
                    if event.data is not None and event.data.get("status") in [
                        "Failed",
                        "Stopped",
                    ]:
                        mon.signal_done()
            assert evaluator._ensemble.get_status() == "Failed"
            assert job_failed
>           assert step_failed
E           assert False

/tmp/jenkins-komodo-f_scout_ci/workspace/unit-tests/ert/test-root/tests/ensemble_evaluator/test_prefect_ensemble.py:709: AssertionError
----------------------------- Captured stdout call -----------------------------
[2021-06-04 14:20:37+0200] ERROR - prefect.0 | This is an expected ERROR
[2021-06-04 14:20:37+0200] ERROR - prefect.TaskRunner | Unexpected error: RuntimeError('This is an expected ERROR',)
Traceback (most recent call last):
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/prefect/engine/runner.py", line 48, in inner
    new_state = method(self, state, *args, **kwargs)
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/prefect/engine/task_runner.py", line 869, in get_task_run_state
    logger=self.logger,
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/prefect/utilities/executors.py", line 299, in run_task_with_timeout
    return task.run(*args, **kwargs)  # type: ignore
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/ert_shared/ensemble_evaluator/entity/function_step.py", line 80, in run
    job=self._step.get_jobs()[0], transmitters=inputs, client=ee_client
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/ert_shared/ensemble_evaluator/entity/function_step.py", line 62, in run_job
    raise e
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/ert_shared/ensemble_evaluator/entity/function_step.py", line 54, in run_job
    output = self._attempt_execute(func=function, transmitters=transmitters)
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/ert_shared/ensemble_evaluator/entity/function_step.py", line 32, in _attempt_execute
    function_output = func(**kwargs)
  File "/tmp/jenkins-komodo-f_scout_ci/workspace/unit-tests/ert/test-root/tests/ensemble_evaluator/test_prefect_ensemble.py", line 660, in function_that_fails_once
    raise RuntimeError("This is an expected ERROR")
RuntimeError: This is an expected ERROR
----------------------------- Captured stderr call -----------------------------
/usr/lib64/python3.6/contextlib.py:88: UserWarning: Tasks were created but not added to the flow: {<Task: Task>}. This can occur when `Task` classes, including `Parameters`, are instantiated inside a `with flow:` block but not added to the flow either explicitly or as the input to another task. For more information, see https://docs.prefect.io/core/advanced_tutorials/task-guide.html#adding-tasks-to-flows.
  next(self.gen)
An exception occurred while starting the ensemble evaluation
Traceback (most recent call last):
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/ert_shared/ensemble_evaluator/prefect_ensemble.py", line 240, in _evaluate
    self.run_flow(ee_id)
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/ert_shared/ensemble_evaluator/prefect_ensemble.py", line 281, in run_flow
    raise state_map[iens].result[task].result
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/prefect/engine/runner.py", line 48, in inner
    new_state = method(self, state, *args, **kwargs)
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/prefect/engine/task_runner.py", line 869, in get_task_run_state
    logger=self.logger,
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/prefect/utilities/executors.py", line 299, in run_task_with_timeout
    return task.run(*args, **kwargs)  # type: ignore
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/ert_shared/ensemble_evaluator/entity/function_step.py", line 80, in run
    job=self._step.get_jobs()[0], transmitters=inputs, client=ee_client
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/ert_shared/ensemble_evaluator/entity/function_step.py", line 62, in run_job
    raise e
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/ert_shared/ensemble_evaluator/entity/function_step.py", line 54, in run_job
    output = self._attempt_execute(func=function, transmitters=transmitters)
  File "/prog/res/komodo/2021.06.rc4-py36-rhel7/root/lib/python3.6/site-packages/ert_shared/ensemble_evaluator/entity/function_step.py", line 32, in _attempt_execute
    function_output = func(**kwargs)
  File "/tmp/jenkins-komodo-f_scout_ci/workspace/unit-tests/ert/test-root/tests/ensemble_evaluator/test_prefect_ensemble.py", line 660, in function_that_fails_once
    raise RuntimeError("This is an expected ERROR")
RuntimeError: This is an expected ERROR
```

--- a/tests/ensemble_evaluator/test_prefect_ensemble.py
+++ b/tests/ensemble_evaluator/test_prefect_ensemble.py
@@ -1,4 +1,5 @@
 from typing import Set
+
 from ert_shared.status.entity.state import JOB_STATE_FAILURE
 import pathlib
 import pickle
@@ -685,9 +686,11 @@ def test_prefect_no_retries(unused_tcp_port, coefficients, tmpdir, function_conf
 
         step_failed = False
         job_failed = False
+        event_list = []
         with evaluator.run() as mon:
             for event in mon.track():
                 # Capture the job error messages
+                event_list.append(event)
                 if event.data is not None and "This is an expected ERROR" in str(
                     event.data
                 ):
@@ -705,8 +708,8 @@ def test_prefect_no_retries(unused_tcp_port, coefficients, tmpdir, function_conf
                 ]:
                     mon.signal_done()
         assert evaluator._ensemble.get_status() == "Failed"
-        assert job_failed
-        assert step_failed
+        assert job_failed, f"Events: {event_list}"
+        assert step_failed, f"Events: {event_list}"
 
 
 def dummy_get_flow(*args, **kwargs):

